lib/client/base.js

```diff
@@ -745,9 +745,16 @@
   HappnClient.prototype.reconnect = function (options) {
     var _this = this;
 
+    if (!this.reconnectAttempt) this.reconnectAttempt = 0;
+    var reconnectAttempt = ++this.reconnectAttempt;
+
     _this.authenticate(function (e) {
+      if (_this.reconnectAttempt != reconnectAttempt) return; // this is not the last attempt, ignore it
 
-      if (e) return _this.handle_error(e, 3);
+      if (e) {
+        if (e.message && e.message.indexOf('api request timed out action') == 0) return _this.reconnect();
```

Comment:
2 comments
- since this is a recursion, probably better to use setImmediate() to limit call stack depth
- is there a possibility of endless retries interfering with shutdown, or is this taken care of at a lower level?